aignostic/aignostic/metrics/metrics.py

```diff
@@ -156,10 +167,58 @@ def macro_recall(name, info: CalculateRequest) -> float:
         ) / len(np.unique(info.true_labels))
 
 
+def _prepare_datasets(info: CalculateRequest):
+    true_labels = np.array(info.true_labels).flatten()
+    predicted_labels = np.array(info.predicted_labels).flatten()
+
+    if not hasattr(info, "protected_attr") or info.protected_attr is None:
+        raise ValueError("protected_attr is missing from the request.")
+
+    protected_attrs = np.array(info.protected_attr).flatten()
+    df_true = pd.DataFrame({'label': true_labels, 'protected_attr': protected_attrs})
+    df_pred = pd.DataFrame({'label': predicted_labels, 'protected_attr': protected_attrs})
+
+    dataset = BinaryLabelDataset(df=df_true, label_names=['label'], protected_attribute_names=['protected_attr'])
+    classified_dataset = BinaryLabelDataset(
+        df=df_pred, label_names=['label'],
+        protected_attribute_names=['protected_attr'])
+    return dataset, classified_dataset
+
+
+def create_fairness_metric_fn(metric_fn):
+    def wrapper(name, info: CalculateRequest) -> float:
+        try:
+            dataset, classified_dataset = _prepare_datasets(info)
+            metric = ClassificationMetric(dataset, classified_dataset,
+                                          privileged_groups=info.privileged_groups,
+                                          unprivileged_groups=info.unprivileged_groups)
+
+            metrics = metric_fn(metric)
+            if (np.isnan(metrics)):
+                metrics = 0
+            return metrics
+        except Exception as e:
+            raise MetricsException(name, additional_context=str(e))
+    return wrapper
+
+
 metric_to_fn = {
     "accuracy": accuracy,
     "class_precision": class_precision,
     "precision": macro_precision,
     "class_recall": class_recall,
-    "recall": macro_recall
+    "recall": macro_recall,
+}
+
+fairness_metrics = {
```

Comment:
There is a faster way to make this using a list comprehension rather than duplication, please do that
(e.g. `fairness_metrics = dict(*[(key, create(lamdba m: m[key]()) for key in some_list])` as a sketch)

Reply:
* more elegant